### ISSUE TITLE
debug level logging, cleanup and usability in tests

### ISSUE THREAD

**Issue by koivunej** [neondatabase/neon#5850]
With `debug` level logging we log too much, however it might allow us to faster to solve regress test failures, so I am proposing that we make `debug` logging less noisy, but still more noisy than `info`.

The same would have to be done if we were to consider per-tenant or per-timeline `debug` or `trace` logging.

_Originally posted by @koivunej in https://github.com/neondatabase/neon/pull/5108#discussion_r1389595844_

**Comment by koivunej** [comment]
@arpad-m noted while discussing the pausable failpoint enhancement #6075 that the meaningless "at failpoint X" are useful tracing when understanding the code and when something happens, so even if they were made conditional on actually pausing, we might consider `debug!` or `trace!` the not paused case.

**Comment by arpad-m** [comment]
Yeah they are "meaningless" in the sense that they don't log anything unexpected, but at the same time we log many such events. It's a question of how much "noise" you want to have in the logs. I personally lean towards a little bit more noise as it might be useful (and for me, during debugging for #5912 it was).
